### DISCUSSION TITLE
How to rotate whole scene

### DISCUSSION THREAD

**Discussion by jcobreros** [NVlabs/instant-ngp#279] [Q&A]
I'm trying to rotate initial camera pose so that the scene aligns better with the unit cube. I've tried modifying the `camera_angle_x` and `camera_angle_y` but I see no change at all, so I guess each picture's `transform_matrix` is fully responsible for the camera pose? Is there another way to do this? My hopes is that by aligning the scene better, it will make marchin cube meshes easier to handle

**Comment by Tom94** [comment]
Hi there, yes, the `transform_matrix` is fully responsible for camera pose -- there's no other way to rotate the scene at the moment.

`camera_angle_x` / `camera_angle_y` refer to the field of view (expressed as an angle) in the x/y direction.

**Comment by jcobreros** [comment]
I see. So if I understood correctly, colmap is responsible for generating every camera pose. And the unit cube is aligned with the pose of the first camera, assuming it has a 45º (approx) vertical angle?

I guess what I'm trying to figure out is how to generate datasets in a way that the unit cube is better aligned with the floor of the scene, so that I can crop the aabb cube and generate more relevant meshes without all the noise from unseen areas.

**Reply by AlbertoOddness** [reply]
Hi! Excuse me to ask on such an old post, but could you figure out if it is the case that the unit cube is aligned with the first camera?

I'm also facing this issue, the unit cube isn't aligned and I'm trying to correct its position so it's easier to generate the aabb cube crops, but so far I haven't really found any good way to do this. If it was related to the first camera though then I'd try maybe setting it to be in the origin and move every other camera the same way
